simphony_mayavi/sources/tests/test_engine_source.py

```diff
@@ -135,9 +135,14 @@ def test_add_to_engine(self):
         # When the source is added to an engine it should load the dataset.
         with self.assertTraitChanges(source, 'data_changed'):
             engine.add_source(source)
-        self.assertIsInstance(source.cuds, Particles)
-        self.assertIsInstance(source._vtk_cuds, VTKParticles)
-        self.assertIsInstance(source.outputs[0], tvtk.PolyData)
+
+        self.assertIsInstance(source.cuds, (Lattice, Particles, Mesh))
+        self.assertIsInstance(source._vtk_cuds, (VTKLattice,
```

Comment:
This change makes the test less strict the original test was verifying that the _vtk_cuds and outputs are of the expected type. Currently we are checking that they are not None (since it cannot be anything else). 

To correct the problem we need one of the following:

- A dictonary that maps the `source` variable to the expected _vtk_cuds and output values
- Set a fixed seed at the start of the test

I prefer the first option